tests/test_pool.py

```diff
@@ -235,6 +236,49 @@ def test_start_methods(self):
                 self.assertIsInstance(results_list, types.GeneratorType)
                 self.assertEqual(self.test_desired_output, sorted(results_list, key=lambda tup: tup[0]))
 
+    def test_mixing_map_calls(self):
+        """
+        When using the same pool, mixing map calls should raise
```

Comment:
It's an implementation thing indeed. It was already not allowed previously (in the current main branch), but I didn't raise an error when it happened, it would just silently fail.

I already have ideas on how to go about allowing this, so I will probably implement that somewhere in the near future. The current PR is actually a good step to allow this.

Reply:
I created a new issue for it: https://github.com/Slimmer-AI/mpire/issues/74